From: Hayes Wang <hayeswang@realtek.com>
Date: Tue, 6 Sep 2011 16:55:16 +0800
Subject: r8169: fix the reset setting for 8111evl
Patch-mainline: v3.1-rc7
Git-commit: c2b0c1e7fb69b54e704cb2dae5a80cc78a8cb0b2
References: bnc#805371

rtl8111evl should stop any TLP requirement before resetting by
enabling register 0x37 bit 7.

Signed-off-by: Hayes Wang <hayeswang@realtek.com>
Acked-by: Michal Kubecek <mkubecek@suse.cz>
---
 drivers/net/r8169.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 5a62862..4e94ed3 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -4223,6 +4223,7 @@ static void rtl8169_hw_reset(struct rtl8169_private *tp)
 		while (RTL_R8(TxPoll) & NPQ)
 			udelay(20);
 	} else if (tp->mac_version == RTL_GIGA_MAC_VER_34) {
+		RTL_W8(ChipCmd, RTL_R8(ChipCmd) | StopReq);
 		while (!(RTL_R32(TxConfig) & TXCFG_EMPTY))
 			udelay(100);
 	} else {
-- 
1.8.1.4

